src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Portable.vb

```diff
@@ -0,0 +1,1522 @@
+
+        '''******************************************************************************
+        ''' ;GetShellPath
+        ''' <summary>
+        ''' Returns the special path format required for pFrom and pTo of SHFILEOPSTRUCT. See NativeMethod.
+        ''' </summary>
+        ''' <param name="FullPaths">A string array containing the paths for the operation.</param>
+        ''' <returns>A string in the required format.</returns>
+        Private Shared Function GetShellPath(ByVal FullPaths() As String) As String
+            ' #If Debug Then
+            Debug.Assert(FullPaths IsNot Nothing, "FullPaths is NULL!!!")
+            Debug.Assert(FullPaths.Length > 0, "FullPaths() is empty array!!!")
+            For Each FullPath As String In FullPaths
+                Debug.Assert(FullPath.Length <> 0 AndAlso IO.Path.IsPathRooted(FullPath), FullPath)
+            Next
+            ' #End If
+            ' Each path will end with a Null character.
+            Dim MultiString As New StringBuilder()
+            For Each FullPath As String In FullPaths
+                MultiString.Append(FullPath & ControlChars.NullChar)
+            Next
+            ' Don't need to append another Null character since String always end with Null character by default.
+            Debug.Assert(MultiString.ToString.EndsWith(ControlChars.NullChar, StringComparison.Ordinal))
+
+            Return MultiString.ToString()
+        End Function
+        Private Shared Function IsEmptyNullOrWhitespace(ByVal StringToCheck As String) As Boolean
+            If StringToCheck Is Nothing Then
+                Return True
+            End If
+            Return StringToCheck.Trim.Length = 0
+        End Function
+
+        '''******************************************************************************
+        ''' ;IsOnSameDrive
+        ''' <summary>
+        ''' Checks to see if the two paths is on the same drive.
+        ''' </summary>
+        ''' <param name="Path1"></param>
+        ''' <param name="Path2"></param>
+        ''' <returns>True if the 2 paths is on the same drive. False otherwise.</returns>
+        ''' <remarks>Just a string comparison.</remarks>
+        Private Shared Function IsOnSameDrive(ByVal Path1 As String, ByVal Path2 As String) As Boolean
+            ' Remove any separators at the end for the same reason in IsRoot.
+            Path1 = Path1.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar)
+            Path2 = Path2.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar)
+            Return String.Compare(IO.Path.GetPathRoot(Path1), IO.Path.GetPathRoot(Path2),
+                    StringComparison.OrdinalIgnoreCase) = 0
+        End Function
+
+        '''**************************************************************************
+        ''' ;IsRoot
+        ''' <summary>
+        ''' Checks if the full path is a root path.
+        ''' </summary>
+        ''' <param name="Path">The p
```

Comment:
HostProtectoinAttribute and all comments about it can be deleted everywhere. Not relevant anymore.